src/main/java/org/graylog/integrations/aws/resources/AWSResource.java

```diff
@@ -77,7 +78,7 @@ public RegionsResponse getAwsRegions() {
     @Path("/available_services")
     @ApiOperation(value = "Get all available AWS services")
     @RequiresPermissions(AWSPermissions.AWS_READ)
-    public AvailableServiceResponse getAvailableServices() {
+    public AvailableServiceResponse getAvailableServices() throws JsonProcessingException {
```

Comment:
JsonProcessingException does return a good response with parsable JSON. An exception mapper is present for this class:

https://github.com/Graylog2/graylog2-server/blob/20ab7325eb82ec09c08ece0433e183412f32a05d/graylog2-server/src/main/java/org/graylog2/shared/rest/exceptionmappers/JsonProcessingExceptionMapper.java#L31

The response looks like this, which seems ok. What do you think?

> < HTTP/1.1 400 Bad Request
> < X-Graylog-Node-ID: e065896b-8a9a-4f45-83f2-e740525ed035
> < X-Runtime-Microseconds: 80559
> < Content-Type: application/json
> < Date: Tue, 16 Jul 2019 15:13:43 GMT
> < Connection: close
> < Content-Length: 236
> < 
> Closing connection 0
>  {"type":"ApiError","message":"No serializer found for class org.graylog.integrations.aws.service.AWSService$1 and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS)"}

Reply:
I am changing my thinking on this. I'm making another change and will post another comment when done.

Reply:
Ok, I have swapped this (in https://github.com/Graylog2/graylog-plugin-integrations/pull/122/commits/51b0a056488678cce042cb9d7169280d728dbe0d) to throw InternalServerErrorException (with a concise, descriptive error message) instead of returning the obscure JSON exception message.

Now the response looks like this: code 400 `{"type":"ApiError","message":"An error occurred encoding the policy JSON"}`